test/Microsoft.AspNet.Mvc.Core.Test/ParameterBinding/ControllerActionArgumentBinderTests.cs

```diff
@@ -343,6 +343,44 @@ public void ParameterHasFieldPrefix([Bind(Prefix = "simpleModelPrefix")] string
             Assert.Null(controller.UnmarkedProperty);
         }
 
+        [Fact]
+        public async Task BindActionArgumentsAsync_DoesNotSetNullValues_ForNonNullableProperties()
+        {
+            // Arrange
+            var actionDescriptor = GetActionDescriptor();
+            actionDescriptor.BoundProperties.Add(
+                new ParameterDescriptor
+                {
+                    Name = "ValueBinderMarkedProperty",
+                    BindingInfo = new BindingInfo(),
+                    ParameterType = typeof(int)
+                });
+
+            var actionContext = GetActionContext(actionDescriptor);
+
+            var binder = new Mock<IModelBinder>();
+            binder
+                .Setup(b => b.BindModelAsync(It.IsAny<ModelBindingContext>()))
+                .Returns(Task.FromResult(
+                    result: new ModelBindingResult(model: null, key: string.Empty, isModelSet: true)));
+            var actionBindingContext = new ActionBindingContext()
+            {
+                ModelBinder = binder.Object,
+            };
+
+            var argumentBinder = GetArgumentBinder();
+            var controller = new TestController();
+
+            // Some non default value.
+            controller.NotNullableProperty = -1;
+
+            // Act
+            var result = await argumentBinder.BindActionArgumentsAsync(actionContext, actionBindingContext, controller);
+
+            // Assert
+            Assert.Equal(-1, controller.NotNullableProperty);
+        }
```

Comment:
We do not have a test for nullable-int, but there are tests for reference type (string). Ill add one for nullable-int as well.